pyro/contrib/gp/models/sgpr.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+from torch.autograd import Variable
+import torch.nn as nn
+
+import pyro
+import pyro.distributions as dist
+
+from .util import _matrix_triangular_solve_compat
+
+
+class _SparseMultivariateNormal(dist.MultivariateNormal):
```

Comment:
Feel free to move this into `pyro.distributions.sparse_mvn` or something. That would make it easier to add distribution tests for it.

Reply:
Sure, after making things work, I will refactor it.